academic-services/knowledge-exploration-service/reference-index-data.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Entity data files
+description: Describes the format for defining entity data to be indexed
+ms.topic: reference
+ms.date: 7/1/2020
+---
+
+# Entity data files
+
+Entity data files define the entity data to be indexed. Each row contains information for a single entity, defined in [JSON format](https://json.org/) and conforming to a specific [schema](reference-index-schema.md) with optional information about its relative probability among other entities.
+
+## Entity data JSON format
+
+Entity data is represented with JSON objects, with attributes values being represented by JSON name/value pairs conforming to the following rules:
+
+Attribute type | JSON value type | Example
+--- | --- | ---
+blob | string | `"Abstract": "It was the best of times, it was the worst of times ... It is a far, far better thing that I do, than I have ever done; it is a far, far better rest that I go to than I have ever known."`
+composite | object | `"F": { "FN": "machine learning", "FId": 119857082 }`
+date | string | `"D": "2015-05-18"`
+double | number | `"logprob": -2.71828182846`
+int32 | number | `"Y": 2015`
+int64 | number | `"Id": 8589934592`
+guid | string | `"Guid": "3d119f86-dcf0-4e35-9c01-b84832d851e9"`
+string | string | `"Ti": "an overview of microsoft academic service mas and applications"`
+
+### Multiple attribute values
+
+To define more than one value for an attribute you can either specify the attribute multiple times, e.g.:
+
+``` JSON
+{ "RId": 123, "RId": 456 }
+{ "F": { "FN": "machine learning", "FId": 119857082 }, "F": { "FN": "artificial intelligence", "FId": 154945302 } }
+```
```

Comment:
These violates JSON's unique key rule. (preprocessor might crash?) Let's not encourage this notation.

Reply:
There is no requirement for JSON object names to be unique:
[https://www.json.org/json-en.html](https://www.json.org/json-en.html)

---
In reply to: [455279249](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/164#discussion_r455279249) [](ancestors = 455279249)

Reply:
Preprocessor won't be able to handle this scenario. It currently depends on newtonsoft to parse JSON and it resolve duplicate keys by taking the last one. I'm not convinced this is a valuable scenario to add support at the moment. 
![image](https://user-images.githubusercontent.com/709064/88216067-70f77f80-cc11-11ea-9f07-ad964f9ec1b8.png)

Reply:
We're not adding support, we're documenting [how KES works](https://github.com/microsoft/cognitive-research-technologies-docs/blob/master/knowledge-exploration-service/DataFormat.md#data-format)